lib/compiler_rt/trunctfxf2.zig

```diff
@@ -49,14 +49,16 @@ pub fn __trunctfxf2(a: f128) callconv(.C) f80 {
         const round_bits = a_abs & round_mask;
         if (round_bits > halfway) {
             // Round to nearest
-            const carry = @boolToInt(@addWithOverflow(u64, res.fraction, 1, &res.fraction));
-            res.exp += carry;
-            res.fraction |= @as(u64, carry) << 63; // Restore integer bit after carry
+            const ov = @addWithOverflow(res.fraction, 1);
+            res.fraction = ov[0];
+            res.exp += ov[1];
+            res.fraction |= @as(u64, ov[1]) << 63; // Restore integer bit after carry
```

Comment:
Ignore this if this was already discussed and firmly agreed on as-is internally but like I said in my comment in the issue (https://github.com/ziglang/zig/issues/10248#issuecomment-1222036225), a regular struct with named fields would be much easier to read here (e.g. `struct { result: T, overflow: u1 }`). Now I'm going to have to memorize what the result value is and what the overflow value is.

These are now the first builtins to my knowledge with this kind of rather not so obvious return type.

If Zig gets destructuring assignment in the future, the situation might improve a bit, though.

Reply:
Just found that functions like `std.math.add`, `std.math.mul`, etc are doing exactly this using builtins as the lower-level construct under the hood. So, for safe math one should just use standard lib functions I guess.